Kinvey.Tests/DataStoreTests/DataStoreAutoIntegrationTests.cs

```diff
@@ -5627,189 +5789,261 @@ public async Task TestSaveAsync()
         }
 
         [TestMethod]
-        public async Task TestSaveMultiInsertNewItemsConnectionAvailableAsync()
+        public async Task TestSaveCreateVersion5ConnectionAvailableAsync()
         {
             // Setup
-            var builder = ClientBuilder.SetFilePath(TestSetup.db_dir);
-
-            if (MockData)
-            {
-                builder.setBaseURL("http://localhost:8080");
-            }
-
-            builder.SetApiVersion("5");
-
-            kinveyClient = builder.Build();
+            kinveyClient = BuildClient("5");
 
             if (MockData)
             {
                 MockResponses(4);
             }
 
             // Arrange
-            var user = await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
-
-            var autoToDoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.AUTO, kinveyClient);
-            var syncToDoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.SYNC, kinveyClient);
+            var autoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.AUTO);
+            var syncStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.SYNC);
+            var networkStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.NETWORK);
 
-            var toDos = new List<ToDo>
+            var newItem = new ToDo
             {
-                new ToDo { Name = "Name1", Details = "Details1", Value = 1 },
-                new ToDo { Name = "Name2", Details = "Details2", Value = 2 }
+                Name = "todo save",
+                Details = "details for save",
+                DueDate = "2016-04-22T19:56:00.961Z"
             };
 
+            await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
+
             // Act
-            var savedToDos = await autoToDoStore.SaveAsync(toDos);
+            var savedItem = await autoStore.SaveAsync(newItem);
 
-            var pendingWriteActions = kinveyClient.CacheManager.GetSyncQueue(toDosCollection).GetAll();
-            var existingToDos = await syncToDoStore.FindAsync();
+            var existingItemCache = await syncStore.FindByIDAsync(savedItem.ID);
```

Comment:
I think it would be better to use just Find and assert the number of items - if for some reason two item is recorded multiple times

Reply:
Ok.